package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,9 +150,34 @@ def parse(self, **kwargs):
         resids = []
         resnames = []
         charges = []
-
+        set_charge = True
+        for line in sections['molecule']:
+            if line.strip().lower() == 'no_charges':
+                set_charge = False
+                break
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
+            columns = a.split()
+            if len(columns) >= 9:
+                columns = columns[:9]
+                aid, name, x, y, z, atom_type, resid, resname, charge = columns
+            elif len(columns) < 6:
+                raise ValueError(f"The @<TRIPOS>ATOM block in mol2 file"
+                                 f" {os.path.basename(self.filename)}"
+                                 f" should contain at least 6 fields to be"
+                                 f" unpacked: atom_id atom_name x y z"
+                                 f" atom_type [subst_id[subst_name"
+                                 f" [charge [status_bit]]]]")
+            else:
+                aid, name, x, y, z, atom_type = columns[:6]
+                resid = '1'
+                resname = ''
```

Comment:
I think @orbeckst said `UNK` was fine here (see https://github.com/MDAnalysis/mdanalysis/issues/3385#issuecomment-1088124131). Any preference @orbeckst ?

Reply:
See my review. After discussion on Discord the consensus is not *not include resname in the Topology* if it's not in the file.